tapdb/supply_ignore_checker.go

```diff
@@ -0,0 +1,224 @@
+package tapdb
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/lightninglabs/taproot-assets/address"
+	"github.com/lightninglabs/taproot-assets/asset"
+	"github.com/lightninglabs/taproot-assets/fn"
+	"github.com/lightninglabs/taproot-assets/proof"
+	"github.com/lightninglabs/taproot-assets/universe/supplycommit"
+	lfn "github.com/lightningnetwork/lnd/fn/v2"
+)
+
+var (
+	// nonGroupedAsset is a zeroed out public key that is used to identify a
+	// non-existent group key (for an asset that does not belong to a
+	// group).
+	nonGroupedAsset = asset.SerializedKey{}
+)
+
+// ignoreCheckerStore is an interface that defines the methods required to
+// fetch supply leaves for the ignore checker. It abstracts away the storage
+// implementation details, simplifying mocking for unit tests.
+type ignoreCheckerStore interface {
+	// FetchSupplyLeavesByType fetches all supply leaves for a given asset
+	// specifier and a specific supply sub-tree.
+	FetchSupplyLeavesByType(ctx context.Context, spec asset.Specifier,
+		tree supplycommit.SupplySubTree, startHeight,
+		endHeight uint32) lfn.Result[supplycommit.SupplyLeaves]
+}
+
+// assetGroupQuery is an interface that defines the method to query an asset
+// group by its asset ID. This is used to determine if an asset belongs to a
+// group and to fetch the group key if it does. It abstracts away the details
+// of how the asset group is stored, simplifying mocking for unit tests.
+type assetGroupQuery interface {
+	// QueryAssetGroupByID attempts to fetch an asset group by its asset ID.
+	// If the asset group cannot be found, then ErrAssetGroupUnknown is
+	// returned.
+	QueryAssetGroupByID(context.Context, asset.ID) (*asset.AssetGroup,
+		error)
+}
+
+// IgnoreCheckerCfg is a configuration struct for the CachingIgnoreChecker.
+type IgnoreCheckerCfg struct {
+	// GroupQuery is used to query asset groups by their asset ID.
+	GroupQuery assetGroupQuery
```

Comment:
Hmm, if the config struct itself is publicly exported, then shouldn't the interfaces exported also be?